drivers/retained_mem/Kconfig

```diff
@@ -15,6 +15,11 @@ config RETAINED_MEM_INIT_PRIORITY
 	help
 	  Retained memory devices initialization priority,
 
+config RETAINED_MEM_INIT_LEVEL_PRE_KERNEL_1
+	bool "Retained memory devices init during PRE_KERNEL_1"
+	help
+	  Retained memory devices initialization during PRE_KERNEL_1 level
```

Comment:
I really do not agree with this, this just makes blind assumptions about drivers which is impossible depending upon the driver

Reply:
If it depends on the driver, that is just that some driver may not be compatible, but I don't see it as a major issue.

Reply:
We could add a new Kconfig symbol, `RETAINED_MEM_SUPPORTS_PRE_KERNEL_INIT` that any compatible drivers could select, and have this symbol depend on that. Likewise, the retention config would also depend on *this* symbol, so the entire "do init early" could only be enabled with a driver that allows it.

Would that address your concerns?